renku_notebooks/api/classes/image.py

```diff
@@ -68,17 +86,43 @@ def get_image_manifest(self, image: "Image") -> Optional[dict[str, Any]]:
         if res.status_code != 200:
             headers["Accept"] = ManifestTypes.oci_v1_index.value
             res = requests.get(image_digest_url, headers=headers)
-            if res.status_code == 200:
-                index_parsed = res.json()
-                manifest = next(
-                    (man for man in index_parsed.get("manifests", []) if man.get("platform", {}).get("os") == "linux"),
-                    None,
-                )
-                manifest = cast(dict[str, Any] | None, manifest)
-                return manifest
         if res.status_code != 200:
             return None
-        return res.json()
+
+        content_type = res.headers.get("Content-Type")
+        if content_type in [ManifestTypes.docker_v2_list.value, ManifestTypes.oci_v1_index.value]:
+            index_parsed = res.json()
+
+            def platform_matches(manifest: dict[str, Any]) -> bool:
+                platform: dict[str, Any] = manifest.get("platform", {})
+                return platform.get("architecture") == platform_architecture and platform.get("os") == platform_os
+
+            manifest: dict[str, Any] = next(filter(platform_matches, index_parsed.get("manifests", [])), {})
+            image_digest: str | None = manifest.get("digest")
+            if not manifest or not image_digest:
+                return None
+            image_digest_url = f"https://{image.hostname}/v2/{image.name}/manifests/{image_digest}"
+            media_type = manifest.get("mediaType")
+            headers["Accept"] = ManifestTypes.docker_v2.value
+            if media_type in [
+                ManifestTypes.docker_v2.value,
+                ManifestTypes.oci_v1_manifest.value,
+            ]:
+                headers["Accept"] = media_type
+            res = requests.get(image_digest_url, headers=headers)
+            if res.status_code != 200:
+                headers["Accept"] = ManifestTypes.oci_v1_manifest.value
+                res = requests.get(image_digest_url, headers=headers)
+            if res.status_code != 200:
+                return None
```

Comment:
nitpick: This whole `if` block should be inside the previous `if` block, since it's acting on the result of the second `get` request on line 115.

Reply:
@m-alisafaee maybe I fail to see it but I think there is no difference between these two code snippets:

```python
            res = requests.get(image_digest_url, headers=headers)
            if res.status_code != 200:
                headers["Accept"] = ManifestTypes.oci_v1_manifest.value
                res = requests.get(image_digest_url, headers=headers)
            if res.status_code != 200:
                return None
```

AND what you propose:

```python
            res = requests.get(image_digest_url, headers=headers)
            if res.status_code != 200:
                headers["Accept"] = ManifestTypes.oci_v1_manifest.value
                res = requests.get(image_digest_url, headers=headers)
                if res.status_code != 200:
                  return None
```

So if the first request is 200 then nothing gets executed from the statements below (in either version). 

If the first request is not 200 then the second request runs for sure. And then the last if statement runs depending on the status code of the second request. The only difference with the suggested edit is that an extra if statement gets tested in the PR version and one less if statement gets tested in the changes you propose (in some cases). So I think it is fine to leave this as is.

Reply:
Ok sorry for the long explanation I see you marked it as nitpick.